src/Interpreters/join_common.cpp

```diff
@@ -597,9 +597,19 @@ NotJoinedBlocks::NotJoinedBlocks(std::unique_ptr<RightColumnsFiller> filler_,
             column_indices_left.emplace_back(left_pos);
     }
 
+    /// `sample_block_names` may contains non unique column names
```

Comment:
> Is it because of aliases?


Kind of, user can specify some column twice (`SELECT a, a FROM table`), or some unexpected case like in test `... RIGHT JOIN ( SELECT count('x'), count('y'), 2 AS x )` will lead to duplicate column with name `count()`